CAIPs/caip-x.md

```diff
@@ -0,0 +1,222 @@
+> Input: name (String) - The domain name to be resolved.
+> Output: Authenticator (String | JSON | null) - A URL or JSON object providing authentication details or additional information, or null if no data can be found.
+
+This function accepts a domain name as a string and returns a URL or a JSON object. The output is intended to provide authentication details or additional information related to the domain name. If no relevant data can be found, the function returns null. This function should be flexible enough to support different formats and data structures, adapting to the needs of various blockchain platforms and applications.
+
+#### ENS example implementation using Typescript and Viem:
+
+```typescript
+import { type Address, type Chain, createPublicClient, http, PublicClient } from "viem";
+import { mainnet } from "viem/chains";
+import { normalize } from "viem/ens";
+
+export interface NameResolver {
+ resolveName(name: string): Promise<Address | null>;
+ resolveAuthenticator(name: string): Promise<string | null>;
+}
+
+export interface ENSOptions {
+ chain?: Chain;
+ jsonRpcUrl?: string;
+}
+
+export class ENS implements NameResolver {
+ private readonly client: PublicClient;
+
+ constructor(options: ENSOptions) {
+   this.client = createPublicClient({
+     chain: options.chain ?? mainnet,
+     transport: http(options.jsonRpcUrl),
+   });
+ }
+
+ async resolveName(domainName: string): Promise<Address | null> {
+   return this.client.getEnsAddress({
+     name: normalize(domainName),
+   });
+ }
+
+ async resolveKey(domainName: string, key: string): Promise<string | null> {
+   return this.client.getEnsText({
+     name: normalize(domainName),
+     key,
+   });
+ }
+
+ async resolveAuthenticator(domainName: string): Promise<string | null> {
+   return this.resolveKey(domainName, "authenticator");
+ }
+}
+```
+
+# Rationale
+
+Specifying the authenticator URL as a domain name NFT text record allows applications to easily discover and integrate with compatible login methods in a standard way
+Having a chain-agnostic standard enables interoperability between different crypto domain providers and authentication methods
+Providing clear wallet implementer steps and code samples makes it easy for developers to adopt this standard
+
+# Backwards Compatibility
+
+This standard is fully backwards compatible as it proposes an additional metadata field for crypto domain NFTs. Existing NFTs and applications will continue to function normally.
+
+# dApp Implementer Steps
+Here are the detailed steps for dApps to integrate Login With Name @domain/loginwithname-wagmi-sdk connector:
+[Login With Name WAGMI SDK](https://github.com/FedericoAmura/login-with-name-wagmi-sdk)
+
+# References
+
+[EIP-4361: Sign-In with Ethereum](https://github.com/ethereum/ercs/blob/master/ERCS/erc-4361.md)
+[Login With Name WAGMI SDK](https://github.com/FedericoAmura/login-with-name-wagmi-sdk)
+
```

Comment:
if you're not familar with this variant of markdown, it's how we do internal links in the CASA docs-- anywhere you put brackets around `ENSIP-5` or `JSON Schema`, it will make a link with these `hrefs`.  see the contributing.md for more details.